### ISSUE TITLE
[FSR] Wizard discovery work

### ISSUE THREAD

**Issue by Dr-Pongo** [department-of-veterans-affairs/va.gov-team#97928]
## Issue Description       
New one thing per page wizard definition includes a fair number of brand new pages all with the same URL root as the FSR. The app generator starts off with the `RoutedSavableApp` definition, so this discovery will mostly be investigating hijacking that to include these new wizard page definitions. 

I'd start by sleuthing through old convos in the following places. They will also be a great source to reach out for input from other devs who may have experimented with this kind of thing in the past
- [#vfs-frontend](https://dsva.slack.com/archives/C5HP4GN3F)
- [#vfs-platform-support](https://dsva.slack.com/archives/CBU0KDSB1)
- [#veteran-facing-forms](https://dsva.slack.com/archives/C044AGZFG2W)

### Helpful links
- **FSR `RoutedSavableApp` definition** 
https://github.com/department-of-veterans-affairs/vets-website/blob/7d67354ed1688d11d8b31d93d8e4747641d3979a/src/applications/financial-status-report/containers/App.jsx#L202-L213

- **[FSR Routes definition](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/financial-status-report/routes.jsx)**
  - maybe we can dump more into `childRoutes`? 

- **[Figma link for new wizard](https://www.figma.com/design/ea8baIDr09ncgr1P4Rfh8i/Financial-Status-Report-(VA5655)---(including-Streamlined-Waiver)?node-id=205-21152&node-type=section&t=wIdtHYu4xshyIXbs-0)**

## Tasks
- [ ] Investigate feasibility for new wizard routing
- [ ] Coordinate with platform for assistance as necessary 
- [ ] Coordinate with design and stakeholders on potential pitfalls
- [ ] Write follow on ticket(s) for implementation
- [ ] Investigate alternatives if this implementation isn't feasible

## Design and Front End sync on implementation of design
- [x] Required
- [ ] Complete
- [ ] Not Required           

## Acceptance Criteria
- [ ] Path forward defined for wizard work
- [ ] Tickets generated for future work if necessary

## Testing
- [x] N/A if non development work
- [ ] Testing passed and documented in this ticket based off the "Then" statement in the description     

## Ticket Creation - Definition of Ready
- [x] Description written
- [x] Tasks defined
- [x] Acceptance criteria written
- [x] Testing defined
- [x] Add Labels
- [x] Attach to an Epic
- [x] Estimate Points

**Comment by josephrlee** [comment]
@Dr-Pongo @amponce, here's the [Mural Link](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1682537441750/a9937e9953d80c221c5165131f611026e0419c7b?sender=u64758e17a18c4eb2bea93888) and a screenshot of the proposed routes for the wizard: 
<img width="1905" alt="Image" src="https://github.com/user-attachments/assets/cd2af347-85c3-488d-ae43-9d2de786298a" />

**Comment by amponce** [comment]
The Wizard will be its own route URL under `manage-va-debt`
 New proposed URL: `manage-va-debt/get-help`

**Comment by amponce** [comment]
Notes on how to set up the new `/debt-help-wizard/ `

Based on:

[Creating a new application](https://depo-platform-documentation.scrollhelp.site/developer-docs/creating-a-new-application
)


### Create a new folder:

`src/applications/debt-help-wizard/
`
### Create manifest.json in that folder:

```
{
  "appName": "Debt Help Options",
  "entryFile": "./debt-help-wizard-entry.jsx",
  "entryName": "debt-help-options",
  "rootUrl": "/manage-va-debt/debt-help-options"
}
```

### Add an entry to `src/applications/registry.json` (in the content-build directory):

```
{
  "appName": "Debt Help Options",
  "entryName": "debt-help-options",
  "rootUrl": "/debt-help-options",
  "template": {
    "title": "Debt Help Options",
    "layout": "page-react.html",
    "description": "Debt help options wizard",
    "keywords": "debt, wizard",
    "vagovprod": false
  }
}
```

Update the vets-api source app middleware configuration to register your new application.
Create the entry file `debt-help-wizard-entry.jsx` in the new folder:

```
import 'platform/polyfills';
import './sass/debt-help-wizard.scss';

import startApp from 'platform/startup';
import routes from './routes';
import reducer from './reducers';
import manifest from './manifest';

startApp({
  url: manifest.rootUrl,
  reducer,
  routes,
  entryName: manifest.entryName,
});
```

Update your revproxy config to include the new application so that internal links via react-router work properly.
Create the Sass file at:

`src/applications/debt-help-wizard/sass/debt-help-wizard.scss`

with the content:

@import "~@department-of-veterans-affairs/formation/sass/shared-variables";

### Create the reducer file at:

```
src/applications/debt-help-wizard/reducers/index.js
```
with the content:
```
const initialState = {};
```

```
function debtHelpOptions(state = initialState, action) {
  switch (action.type) {
    case 'SOME_ACTION':
      return state;
    default:
      return state;
  }
}

export default {
  debtHelpOptions
};
```

### Create the routes file at:

`src/applications/debt-help-wizard/routes.jsx`

with the content:

```
import React from 'react';
import { Route, IndexRedirect } from 'react-router';

import DebtHelpWizardApp from './containers/DebtHelpWizardApp.jsx';
import DebtHelpOptionsPage from './containers/DebtHelpOptionsPage.jsx';

const routes = (
  <Route path="/" component={DebtHelpWizardApp}>
    <Route path="debt-page" component={DebtHelpOptionsPage} />
    <IndexRedirect to="debt-page" />
  </Route>
);

export default routes;
```

PR Merge Order: First merge the vets-website PR, then the content-build PR. Once the vets-website PR is merged and CI on main completes (making the build assets available), the automated tests for the registry addition in the content-build PR should pass.

Going to Production: Update the registry entry to set `"vagovprod": true:`

```
{
  "appName": "Debt Help Options",
  "entryName": "debt-help-options",
  "rootUrl": "/debt-help-options",
  "template": {
    "title": "Debt Help Options",
    "layout": "page-react.html",
    "description": "Debt help options wizard",
    "keywords": "debt, wizard",
    "vagovprod": true
  }
}
```

### Directory Structure:

```
/manage-va-debt/          // Debt management landing page

/debt-help-options/        // Wizard routes
  ├── /debt-help-options/intro/
  ├── /debt-help-options/[questionId]/
  ├── /debt-help-options/results/
  └── /debt-help-options/5655/

```

**Comment by amponce** [comment]
thanks for the awesome examples and notes @Dr-Pongo

**Comment by josephrlee** [comment]
@amponce This looks correct, but let me confirm the root URL `/manage-va-debt/debt-help-options`.

cc: @mgleo

**Comment by mgleo** [comment]
Hi @amponce 
We won't be moving the FSR. We're getting [confirmation from CAIA](https://github.com/department-of-veterans-affairs/va.gov-team/issues/69414#issuecomment-2637048493) on the directory structure and naming. Stay tuned! 


cc @josephrlee
